docs/rfcs/20220829-distributed-dynamic-routing.md

```diff
@@ -0,0 +1,224 @@
+- Feature Name: distributed-dynamic-routing
+- Tracking Issue: [#198](https://github.com/CeresDB/ceresdb/issues/198)
+
+# Summary
+Distributed dynamic routing for CeresDB cluster mode,  provide failover and load balancing for cluster.
+
+# Motivation
+In Release v0.3, we initially implemented the cluster mode of CeresDB and provided static routing based on configuration.
+However, it is obvious that this simple implementation cannot support the dynamic scheduling in distributed scenarios.
+Through distributed dynamic routing, we hope to achieve：
+
+- Provide disaster recovery and high availability for the distributed mode of CeresDB.
+- Control the scheduling of CeresDB computing nodes and provide good load balancing capability. Automatically schedule the routing relationship between data and computing nodes to solve the hot issues of computing nodes
+
+# Details
+### Cluster Mode Architecture
+```
+                                               ┌──────────────────┐                                                  
+                           ┌───────────────────┤CeresMeta Cluster ├────────────────────┐                             
+                           │                   └──────────────────┘                    │                             
+                           │     ┌────────┐    ┌────────┐   ┌────────┐   ┌────────┐    │                             
+                           │     │        │    │        │   │        │   │        │    │                             
+                           │     │  etcd  │    │  etcd  │   │  etcd  │   │  etcd  │    │                             
+                           │     │        │    │        │   │        │   │        │    │                             
+                           │     └────────┘    └────────┘   └────────┘   └────────┘    │                             
```

Comment:
Nit-picking: as a raft cluster, it's more general that containing an odd number of instances, like 3 or 5

Reply:
Yes, an odd number of nodes can better reflect the deployment structure of the cluster.